packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+          address: "avalidemailaddressfortenant@testingagreement.com",
+          createdAt: new Date(),
+        },
+      ],
+    };
+
+    const descriptor = {
+      ...getMockDescriptor(),
+      state: descriptorState.published,
+    };
+    const eservice = getMockEService(generateId<EServiceId>(), producer.id, [
+      descriptor,
+    ]);
+
+    const agreement = {
+      ...getMockAgreement(eservice.id, consumer.id),
+      producerId: producer.id,
+      descriptorId: generateId<DescriptorId>(),
+    };
+
+    await addOneEService(eservice);
+    await addOneTenant(consumer);
+    await addOneTenant(producer);
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(notLatestEServiceDescriptor(agreement.descriptorId));
+  });
+
+  it("should throw a descriptorNotInExpectedState error when eservice latest descriptor has invalid state", async () => {
+    const producer = getMockTenant();
+    const consumer = {
+      ...getMockTenant(),
+      mails: [
+        {
+          id: generateId(),
+          kind: tenantMailKind.ContactEmail,
+          address: "avalidemailaddressfortenant@testingagreement.com",
+          createdAt: new Date(),
+        },
+      ],
+    };
+
+    const allowedStatus: DescriptorState[] = [
+      descriptorState.published,
+      descriptorState.suspended,
+    ];
+    const descriptor = {
+      ...getMockDescriptor(),
+      state: randomArrayItem(
+        Object.values(descriptorState).filter(
+          (state: DescriptorState) =>
+            !allowedStatus.includes(state) && state !== descriptorState.draft
+        )
+      ),
+    };
+    const eservice = getMockEService(generateId<EServiceId>(), producer.id, [
+      descriptor,
+    ]);
+
+    const agreement = {
+      ...getMockAgreement(eservice.id, consumer.id),
+      producerId: producer.id,
+      descriptorId: eservice.descriptors[0].id,
+    };
+
+    await addOneEService(eservice);
+    await addOneTenant(consumer);
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(
+      descriptorNotInExpectedState(eservice.id, descriptor.id, allowedStatus)
+    );
+  });
+
+  it("should throw a tenantNotFound error when producer does not exist in tenant collection", async () => {
```

Comment:
Add the same test also for the case where consumer does not exist (it should probably be before the `validateConsumerEmail` test)

Reply:
Adde with 👉🏻 [9c47362065f7ef838071fa6c10b8628838e930a5](https://github.com/pagopa/interop-be-monorepo/pull/603/commits/9c47362065f7ef838071fa6c10b8628838e930a5)